yt/fields/vector_operations.py

```diff
@@ -599,6 +599,71 @@ def _cartesian_z(field, data):
                 display_field=True,
             )
 
+    if registry.ds.geometry == "spherical":
+
+        def _cylindrical_radius_component(field, data):
+            return data[(ftype, f"{basename}_r")] * np.sin(data[(ftype, "theta")])
+
+        registry.add_field(
+            (ftype, f"{basename}_cylindrical_radius"),
+            sampling_type="local",
+            function=_cylindrical_radius_component,
+            units=field_units,
+            display_field=True,
+        )
+
+        registry.alias(
+            (ftype, f"{basename}_cylindrical_z"),
+            (ftype, f"{basename}_cartesian_z"),
+        )
+
+        def _poloidal_magnitude(field, data):
+            return np.sqrt(
+                data[(ftype, f"{basename}_r")] ** 2
+                + data[(ftype, f"{basename}_theta")] ** 2
+            )
+
+        registry.add_field(
+            (ftype, f"{basename}_poloidal_magnitude"),
+            sampling_type="local",
+            function=_poloidal_magnitude,
+            units=field_units,
+            display_field=True,
+        )
+
+        registry.alias(
+            (ftype, f"{basename}_toroidal_magnitude"),
+            (ftype, f"{basename}_phi"),
+        )
```

Comment:
Two notes here:
1) these are not necessary for the rest of the PR, I'm just throwing them in while I'm here
2) I'm deliberately not following the existing naming convention for "magnetic_field_poloidal" and "magnetic_field_toroidal" fields that are defined in a field plugin, because their definition is not robust ("bulk magnetic field" is a requirement) and they live in the plugin system, which is fragile, so I prefer setting solid defaults here, but I'll avoid name collisions.

Reply:
The plugin system is fragile?

Also, I thought that bulk magnetic field had a default value of 0?

Reply:
I'm saying it's fragile because, AFICT, the field plugins rely on yt.mods, which has has comments saying it's dangerous, and is also deprecated.

I wasn't aware that there was a default bulk magnetic field however.

Reply:
I'll take that back actually:
- `"bulk_magnetic_field"` field parameter is indeed not "required" and correctly set to 0 by default.
- it doesn't matter how fragile the field plugin system is, as it's still part of things we want to continue supporting. So either we'll come up with a smart way to maintain backward compatibility via a refactor, or we can just _not_ follow through yt.mods' deprecation, if that turn out impossible.

I'll lighten this PR and removing these concurrent definitions.

Reply:
Turns out it's harder than I thought, see #4020
I may just remove this PR from the 4.1 milestone if I don't make fast enough progress

Reply:
actually solving this much quicker than I thought 20min ago: #4021